transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java

```diff
@@ -35,6 +36,22 @@
     private final String name;
     private boolean removed;
 
+    /**
+     * This is set to {@code true} once the {@link ChannelHandler#handlerAdded(ChannelHandlerContext) method was called.
+     * We need to keep track of this This will set to true once the
+     * {@link ChannelHandler#handlerAdded(ChannelHandlerContext)} method was called. We need to keep track of this
+     * to ensure we will never call another {@link ChannelHandler} method before handlerAdded(...) was called
+     * to guard againstordering issues. {@link ChannelHandler#handlerAdded(ChannelHandlerContext)} MUST be the first
+     * method that is called for handler when it becomes a part of the {@link ChannelPipeline} in all cases. Not doing
+     * so may lead to unexpected side-effects as {@link ChannelHandler} implementationsmay need to do initialization
+     * steps before a  {@link ChannelHandler} can be used.
+     *
+     * See <a href="https://github.com/netty/netty/issues/4705">#4705</a>
+     *
+     * No need to mark volatile as this will be made visible as next/prev is volatile.
+     */
+    private boolean handlerAdded;
```

Comment:
Let me rebame to handlerRemoved

> Am 19.01.2016 um 02:15 schrieb Trustin Lee <notifications@github.com>:
> 
> In transport/src/main/java/io/netty/channel/AbstractChannelHandlerContext.java:
> 
> > @@ -35,6 +36,22 @@
> >      private final String name;
> >      private boolean removed;
> >  
> > +    /**
> > +     * This is set to {@code true} once the {@link ChannelHandler#handlerAdded(ChannelHandlerContext) method was called.
> > +     * We need to keep track of this This will set to true once the
> > +     * {@link ChannelHandler#handlerAdded(ChannelHandlerContext)} method was called. We need to keep track of this
> > +     * to ensure we will never call another {@link ChannelHandler} method before handlerAdded(...) was called
> > +     * to guard againstordering issues. {@link ChannelHandler#handlerAdded(ChannelHandlerContext)} MUST be the first
> > +     * method that is called for handler when it becomes a part of the {@link ChannelPipeline} in all cases. Not doing
> > +     * so may lead to unexpected side-effects as {@link ChannelHandler} implementationsmay need to do initialization
> > +     * steps before a  {@link ChannelHandler} can be used.
> > +     *
> > +     * See <a href="https://github.com/netty/netty/issues/4705">#4705</a>
> > +     *
> > +     * No need to mark volatile as this will be made visible as next/prev is volatile.
> > +     */
> > +    private boolean handlerAdded;
> Another nit: we have removed field above. Should it be renamed to handlerRemoved or should this field be renamed to added?
> 
> —
> Reply to this email directly or view it on GitHub.
>